src/teamforge.iced

```diff
@@ -0,0 +1,371 @@
+{make_esc} = require 'iced-error'
+{athrow,akatch,unix_time} = require('iced-utils').util
+kbpgp = require 'kbpgp'
+proofs = require 'keybase-proofs'
+constants = proofs.constants
+{prng,createHash} = require 'crypto'
+{PerUserSecretKeys, PerTeamSecretKeys, PerTeamKeyBoxes, PerTeamSecretKeySet, PerTeamKeyBox} = require './teamlib'
+
+#===================================================
+
+assert = (condition, msg) ->
+  unless condition
+    throw new Error msg
+
+#===================================================
+
+exports.TeamForge = class TeamForge
+
+  #-------------------
+
+  constructor : ({@chain}) ->
+
+  #-------------------
+
+  forge : (cb) ->
+    esc = make_esc cb, "TeamForge::forge"
+
+    # temporary state for the forge
+    @teams = {}
+    @users = {}
+    # return value for the forge
+    @out =
+      log: []
+      teams: {}
+      users: {}
+      key_owners: {} # map from keyid to user_label
+      key_pubkeyv2nacls: {} # map from keyid to keybase1.PublicKeyV2NaCl's
+      team_merkle: {} # map from team id to {seqno, linkid}
+      merkle_triples: {} # map from keys with a dash "LeafID-HashMeta" to MerkleTriple's
+      expect: @chain.expect
+    @_link_id_gen = new LinkIDGen
+
+    # This is the user who is loading teams.
+    # They are also the default for forging actions when no one else is specified.
+    @default_user_label = "herb"
+
+    for user_label, user_desc of @chain.users
+      await user = new User {forge:@, label: user_label}, esc defer()
```

Comment:
as a convention I've adopted, I never make constructors async. Can you do something like this instead?

```iced
class User
   @make : ({forge, label}, cb) ->
     user = new User { forge, label }
      # blah blah
      cb null, user

await User.make { forge : @, label : user_label }, esc defer user
```